### PR TITLE
Feature/zcs 3128

### PR DESCRIPTION
**Pull request by grishick**
With these changes, the extension can be used for testing. What's left is more testing for followup synchronization.

### CODE REVIEW

**Review by tmclane — APPROVED** [review]

**Review by desouzas — APPROVED** [review]

File: src/java/com/synacor/zimbra/ys/contacts/YahooContactsImport.java [hunk 1]
```diff
@@ -182,8 +189,76 @@ public void test() throws ServiceException {
...
+        }
+        HttpGet get = new HttpGet(String.format(YSocialURLPattern, tokenAndGuid.getSecond(), "json", rev));
+        String authorizationHeader = String.format("Bearer %s", tokenAndGuid.getFirst());
+        get.addHeader(HttpHeaders.AUTHORIZATION, authorizationHeader);
+        HttpClient client = ZimbraHttpClientManager.getInstance().getExternalHttpClient();
+        JsonArray jsonContacts = null;
+        HttpResponse response = null;
+        String respContent = "";
+        try {
+            response = client.execute(get);
```
Comment (fatkudu) [inline src/java/com/synacor/zimbra/ys/contacts/YahooContactsImport.java]: Is there a way to log the request and response should that be desired?

File: src/java/com/synacor/zimbra/ys/contacts/YahooContactsUtil.java [hunk 2]
```diff
@@ -327,61 +344,77 @@ public static ParsedContact parseYContact(JsonObject jsonContact, DataSource ds)
...
-                                Locale locale = null;
-                                if(ds != null) {
-                                    locale = ds.getAccount().getLocale();
+                            String fieldType = fieldObj.get(TYPE).getAsString().toLowerCase();
+                            YahooContactsUtil.YContactFieldType type = null;
+                            if(fieldType != null && !fieldType.isEmpty()) {
+                                try {
+                                    type = YahooContactsUtil.YContactFieldType.valueOf(fieldType);
+                                } catch(Exception e) {
+                                    ZimbraLog.extensions.warn("YahooContactsUtil cannot map Yahoo Contact field of type '%s' to any Zimbra contact field. Ignoring the field.", fieldType);
```
Comment (fatkudu) [inline src/java/com/synacor/zimbra/ys/contacts/YahooContactsUtil.java]: I'm sure contacts has a way to store unknown data with some kind of key=value mechanism??

**Review by fatkudu — APPROVED** [review]
Looks good.  Couple of queries about possible future improvements.

**Review by fatkudu — CHANGES_REQUESTED** [review]
If you export your contacts to CSV, then edit the target file and add some randomly named fields at the end - say "Wibble" and "Wobble", then add some values for those fields for a contact - say - "Potato" and "Waffles", then import that into another account, the UI shows those fields in an Other category, with the values in one column and field name in another.  This way you preserve both the old field name and the value rather than just the value.

**Pull request merged**